### PR TITLE
Feat: Show monitor ID in Details

### LINKED ISSUE #4274: View monitor id
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

API, UI Feature, Other

### 🔖 Feature description

I'm using an uptime-kuma-api made with websocket, as there isn't an "official" one yet. The problem is that I need to get the monitor's id to be able to get its result, and so far I haven't found any place in the interface where I could get this id. This would certainly be easy for me, but making a request for all monitors to find the ID would be bad for my client.

### ✔️ Solution

I propose put a id in monitor infos

### ❓ Alternatives

_No response_

### 📝 Additional Context

_No response_

### PR DESCRIPTION
**Pull request by chakflying**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

# Description

Fixes #4274

## Type of change

- User interface (UI)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [ ] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

image

### CODE REVIEW

**Review by Zaid-maker — APPROVED** [review]
Amazing feature

**Comment by CommanderStorm** [comment]
I am not quite convinced the current design fits as is.
I think adding this as a tag (like "web" in the screenshot) or as as a grayer, smaller #ID next to the tilte could work better. Have you experimented with these approaches?

File: src/pages/Details.vue [hunk 1]
```diff
@@ -4,8 +4,11 @@
             <router-link v-if="group !== ''" :to="monitorURL(monitor.parent)"> {{ group }}</router-link>
             <h1> {{ monitor.name }}</h1>
             <p v-if="monitor.description">{{ monitor.description }}</p>
-            <div class="tags">
-                <Tag v-for="tag in monitor.tags" :key="tag.id" :item="tag" :size="'sm'" />
+            <div class="d-flex">
+                <span class="me-2 my-1 monitor-id">ID: {{ monitor.id }}</span>
```
Comment (Saibamen) [inline src/pages/Details.vue:8]: Suggestion:
```suggestion
                <span class="me-2 my-1 monitor-id">#{{ monitor.id }}</span>
```

**Comment by chakflying** [comment]
Screenshot 2024-01-06 050644
Screenshot 2024-01-10 022450
Screenshot 2024-01-10 022543
Screenshot 2024-01-10 023628

Personally, I don't like the "Tag that is not a Tag" design, and the font size had to be further reduced to fit in the same layout.
Putting it next to the title looks fine, but because the title is variable length, the location would move around, which is not convenient for finding it, and we have to deal with overflow.
Using `#` is also undesirable since it makes selecting only the ID for copying harder.

**Comment by Saibamen** [comment]
>Using # is also undesirable since it makes selecting only the ID for copying harder.

What? Just double-click on number: #1421343224

This is what will be selected - no `#` in selection:

image

**Comment by Saibamen** [comment]
Or you can implement copy ID button :)

**Comment by CommanderStorm** [comment]
> since it makes selecting only the ID for copying harder

Have not tested it, but I think this can be disabled: https://dev.to/smpnjn/how-to-disable-text-selection-with-css-4jan

**Comment by chakflying** [comment]
> > Using # is also undesirable since it makes selecting only the ID for copying harder.
> 
> What? Just double-click on number: #1421343224
> 
> This is what will be selected - no `#` in selection:
> 
> image

Unfortunately does not work on Firefox Mobile, I guess it depends on browser support.

Using CSS to disable selection is a valid option I guess.

**Comment by louislam** [comment]
> Screenshot 2024-01-10 022450

I think I may prefer this, but with smaller font size. Because it does not require a new line when there are no tags.

For highlight issue, I think `inline-block` should do the trick.
https://jsfiddle.net/louislam/cLw0rbxy/

**Comment by chakflying** [comment]
Updated design according to suggestions.

image

image

**Review by CommanderStorm — APPROVED** [review]

**Pull request merged**